coalib/output/Logging.py

```diff
@@ -4,25 +4,48 @@
 import logging
 import logging.config
 
+from coalib.output.printers.LOG_LEVEL import LOG_LEVEL
 
-def configure_logging():
+
+def configure_logging(new_log_level=logging.INFO, incremental=False):
     """
     Configures the logging with hard coded dictionary.
+
+    If the incremental param is ``True``, all the log-handlers configuration
+    other than level will be ignored (including class).
+    Thus, use it only when changing the log_level.
+    When changing log_level, ``json-handler``, ``console-handler``,
+    and ``file-handler`` should all exist.
+    ``json-handler`` is created as a null handler to avoid this error.
+
+    :param new_log_level:
+        The new log level to set for json and console handlers.
+    :param incremental:
+        ``False`` if setting up new loggers, ``True`` if only changing
+        the log level.
+    :return:
+        The new_log_level is returned.
     """
     import sys
 
     logging.config.dictConfig({
         'version': 1,
+        'incremental': incremental,
         'handlers': {
             'console-handler': {
                 'class': 'logging.StreamHandler',
                 'formatter': 'color',
-                'stream': sys.stderr
+                'stream': sys.stderr,
+                'level': LOG_LEVEL.reverse.get(new_log_level)
```

Comment:
Initially we were changing the log level of the only logger we are using.

How Python Logging works is that we give a message to a logger, 
The message is filtered first by the log level of the **logger**.
The log level of the **handler** comes into play later.

So instead of changing the log level of the logger as we used to do, I am changing the log level of the Handler(s).

![image](https://cloud.githubusercontent.com/assets/15715894/24077530/228fec96-0c76-11e7-8810-49b9ca3d730c.png)